cypress/e2e/session-recording.cy.js

```diff
@@ -52,24 +71,230 @@ describe('Session recording', () => {
+                        expect(c.properties['$session_id']).to.equal(sessionId)
+                    })
+                    expect(sessionId).not.to.be.null
+                })
+            })
+            // and then reset
+            cy.resetPhCaptures()
+            // and refresh the page
+            cy.reload()
+            onPageLoad()
+
+            cy.get('body')
+                .trigger('mousemove', { clientX: 200, clientY: 300 })
+                .trigger('mousemove', { clientX: 210, clientY: 300 })
+                .trigger('mousemove', { clientX: 220, clientY: 300 })
+                .trigger('mousemove', { clientX: 240, clientY: 300 })
+
+            cy.wait('@session-recording').then(() => {
+                cy.phCaptures({ full: true }).then((captures) => {
+                    // should be a $snapshot for the current session
+                    expect(captures.map((c) => c.event)).to.deep.equal(['$pageview', '$snapshot'])
+                    expect(captures[0].properties['$session_id']).to.equal(sessionId)
+                    expect(captures[1].properties['$session_id']).to.equal(sessionId)
+
+                    // the amount of captured data should be deterministic
+                    // but of course that would be too easy
+                    expect(captures[1]['properties']['$snapshot_data']).to.have.length.above(0)
+
+                    /**
+                     * the snapshots will look a little like:
+                     * [
+                     *  {"type":3,"data":{"source":6,"positions":[{"x":58,"y":18,"id":15,"timeOffset":0}]},"timestamp":1699814887222},
+                     *  {"type":3,"data":{"source":6,"positions":[{"x":58,"y":18,"id":15,"timeOffset":-430}]},"timestamp":1699814887722}
+                     *  ]
+                     */
+
+                    // page reloaded so we will start with a full snapshot
+                    // a meta and then a full snapshot
+                    expect(captures[1]['properties']['$snapshot_data'][0].type).to.equal(4) // meta
+                    expect(captures[1]['properties']['$snapshot_data'][1].type).to.equal(2) // full_snapshot
+
+                    const xPositions = []
+                    for (let i = 2; i < captures[1]['properties']['$snapshot_data'].length; i++) {
+                        expect(captures[1]['properties']['$snapshot_data'][i].type).to.equal(3)
+                        expect(captures[1]['properties']['$snapshot_data'][i].data.source).to.equal(
+                            6,
+                            JSON.stringify(captures[1]['properties']['$snapshot_data'][i])
+                        )
+                        xPositions.push(captures[1]['properties']['$snapshot_data'][i].data.positions[0].x)
+                    }
+
+                    // even though we trigger 4 events, only 2 snapshots should be captured
+                    // I _think_ this is because Cypress is faking things and they happen too fast
```

Comment:
```suggestion
                    // even though we trigger 4 events, only 2 snapshots should be captured
                    // This is because rrweb doesn't try to capture _every_ mouse move
```